source/gfoidl.DataCompression/Compression/DeadBandCompression.cs

```diff
@@ -62,114 +65,296 @@ public DeadBandCompression(double instrumentPrecision, in TimeSpan maxTime)
         /// <summary>
         /// Implementation of the compression / filtering.
         /// </summary>
+        /// <typeparam name="TList">The type of the enumeration / list.</typeparam>
         /// <param name="data">Input data</param>
         /// <returns>The compressed / filtered data.</returns>
         protected override IEnumerable<DataPoint> ProcessCore(IEnumerable<DataPoint> data)
         {
-            if (data is IList<DataPoint> list) return this.ProcessCore(list);
+            if (data is ArrayWrapper<DataPoint> arrayWrapper)
+            {
+                return arrayWrapper.Count == 0
+                    ? DataPointIterator.Empty
+                    : new IndexedIterator<ArrayWrapper<DataPoint>>(this, arrayWrapper);
+            }
+
+            if (data is ListWrapper<DataPoint> listWrapper)
+            {
+                return listWrapper.Count == 0
+                    ? DataPointIterator.Empty
+                    : new IndexedIterator<ListWrapper<DataPoint>>(this, listWrapper);
+            }
+
+            if (data is DataPoint[] array)
+            {
+                return array.Length == 0
+                    ? DataPointIterator.Empty
+                    : new IndexedIterator<ArrayWrapper<DataPoint>>(this, new ArrayWrapper<DataPoint>(array));
+            }
 
-            return this.ProcessCore(data.GetEnumerator());
+            if (data is List<DataPoint> list)
+            {
+                return list.Count == 0
+                    ? DataPointIterator.Empty
+                    : new IndexedIterator<ListWrapper<DataPoint>>(this, new ListWrapper<DataPoint>(list));
+            }
+
+            IEnumerator<DataPoint> enumerator = data.GetEnumerator();
```

Comment:
`IList<DataPoint>` can be handled the same way as list and array (w/o wrapper).

Add tests for that case.